src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -659,7 +660,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 {
                     countOfExplicitNames++;
 
-                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.UnderlyingMemberName(i))
+                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.TupleMemberName(i))
```

Comment:
We have to allow `Rest` at position 8 (and make it function properly, meaning it would return the `Rest` field without unwrapping if it is a `ValueTuple`) for compatibility with C# 6 users of `ValueTuple`.